File: algorithms/Deque/Easy/StackCRUDLinkedList/VSDStackCRUDLinkedList.java
// Copyrights to venkys.io
// For more programs visit venkys.io 
// Java program for Stack

class Node {
    int data;
    Node next;

    Node(int data) {
        this.data = data;
    }
}

class Stack {
    Node top;

    void push(int data) {
        if (top == null) {
            top = new Node(data);
            return;
        }
        Node temp = new Node(data);
        temp.next = top;
        top = temp;

    }

    void pop() {
        if (top == null)
            return;
        top = top.next;
    }

    void peek() {
        if (top == null)
            return;
        System.out.println(top.data);
    }

    void print() {
        Node temp = top;
        while (temp != null){
            System.out.print(temp.data + " ");
            temp=temp.next;
        }
        System.out.println();
    }
}

class Main {
    public static void main(String[] args) {
        Stack s = new Stack();
        s.push(10);
        s.push(20);
        s.peek();
        s.print();
        s.pop();
        s.push(30);
        s.push(40);
        s.peek();
        s.print();

    }
}

File: algorithms/Deque/Easy/StackCRUDLinkedList/VSDStackCRUDLinkedList.py
# // Copyrights to venkys.io
# // For more programs visit venkys.io 
# // Java program for Stack
class Node:
    def __init__(self,data):
        self.data=data 
        self.next=None 

class StackLL:
    def __init__(self):
        self.top=None 

    
    def isempty(self):
        return not self.top 

    def push(self,data):
        if not self.top:
            self.top=Node(data)
        else:
            temp=Node(data)
            temp.next=self.top
            self.top=temp 
    
    def pop(self):
        if self.top:
            x=self.top.data 
            self.top=self.top.next 
            return x
    
    def traverse(self):
        head=self.top
        while head:
            print(head.data,end=" ")
            head=head.next 
        print()
    

stack=StackLL()
stack.push(10)
stack.push(20)
stack.push(30)
stack.traverse()
stack.push(40)
stack.push(50)
stack.push(60)
stack.traverse()
stack.pop()
stack.pop()
stack.pop()
stack.traverse()